symengine/complex.h

```diff
@@ -64,15 +64,14 @@ class Complex : public Number {
     static RCP<const Number> from_two_nums(const Number &re,
         const Number &im);
 
+    //! \return `true` if exactly `0`
+    virtual bool is_exact_zero() const { return false; }
     //! \return `true` if `0`
-    virtual bool is_zero() const { return ((this->real_ == 0) && (this->imaginary_ == 0)); }
+    virtual bool is_zero() const { return false; }
```

Comment:
I know there are there, but if we later decide to change it, then these routines like `is_zero` start failing. I think it's a good habit to check these things at the point when they are needed as well. Though you are right, that we mostly do not check what we assert in the canonicalization again. So either is fine with me. Thanks for adding the comment.